internal/provider/common.go

```diff
@@ -55,7 +55,7 @@ func isNodeUp(ctx context.Context, nc *client.NodeClient) error {
 	return nil
 }
 
-func isNodesUp(ctx context.Context, sub subi.SubstrateExt, nodes []uint32, nc client.NodeClientCollection) error {
+func isNodesUp(ctx context.Context, sub *substrate.Substrate, nodes []uint32, nc client.NodeClientCollection) error {
```

Comment:
`areNodesUp`, also the internal code can be done in separate goroutines  and wait on them